pkg/core/resource_graph.go

```diff
@@ -135,17 +119,48 @@ func (rg *ResourceGraph) AddDependenciesReflect(source Resource) {
 		case reflect.Slice, reflect.Array:
 			for elemIdx := 0; elemIdx < fieldValue.Len(); elemIdx++ {
 				elemValue := fieldValue.Index(elemIdx)
-				add(elemValue)
+				rg.addDependenciesReflect(source, elemValue)
```

Comment:
The godoc for this function needs to be updated to reflect (heh) that it is recursive and not just first-level anymore.